### PR TITLE
feat: full symbolic evaluation of `gcm_gmult_v8_program`

### PR DESCRIPTION
**Pull request by alexkeizer**
### Description:

This PR adds a few SIMD instructions to the `state_simp_rules` simpset, 
and changes `LNSymSimpContext` to also load simprocs which are part of the passed simpsets (previously only default simprocs were being used).

Together, this enables full symbolic evaluation of `gcm_gmult_v8`

### Testing:

`make all` succeeds locally. Conformance testing failed, but this issue was actually present in main already (and will be fixed by #61)

### License:

By submitting this pull request, I confirm that my contribution is
made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: Tests/AES-GCM/GCMGmultV8Sym.lean
Comment (shigoel) [inline Tests/AES-GCM/GCMGmultV8Sym.lean:1]: Let's create a new directory under `Proofs`, say `Proofs/AES_GCM`, and move this file there.

`Tests` directory pertains to concrete tests.

Comment (alexkeizer) [reply Tests/AES-GCM/GCMGmultV8Sym.lean:1]: Done, I've also added it to the import path of `Proofs.lean`, so that it'll get built by CI

**Comment by alexkeizer** [comment]
Something funny is going on here: CI fails, yet both `make all` and opening the failing file directly in VSCode works fine

**Review by shigoel — APPROVED** [review]
Thanks, Alex!

**Pull request merged**